{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"final_flask_app.ipynb","provenance":[],"collapsed_sections":[],"authorship_tag":"ABX9TyOKL23AdEGepHeGAu+DdySS"},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"},"accelerator":"GPU"},"cells":[{"cell_type":"code","metadata":{"id":"x--Enl_28rgJ","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1620213779687,"user_tz":-330,"elapsed":4852,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"94642c83-197d-4ac5-8ace-4cd026ba26df"},"source":["!pip install flask-ngrok"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Collecting flask-ngrok\n","  Downloading https://files.pythonhosted.org/packages/af/6c/f54cb686ad1129e27d125d182f90f52b32f284e6c8df58c1bae54fa1adbc/flask_ngrok-0.0.25-py3-none-any.whl\n","Requirement already satisfied: Flask>=0.8 in /usr/local/lib/python3.7/dist-packages (from flask-ngrok) (1.1.2)\n","Requirement already satisfied: requests in /usr/local/lib/python3.7/dist-packages (from flask-ngrok) (2.23.0)\n","Requirement already satisfied: itsdangerous>=0.24 in /usr/local/lib/python3.7/dist-packages (from Flask>=0.8->flask-ngrok) (1.1.0)\n","Requirement already satisfied: Werkzeug>=0.15 in /usr/local/lib/python3.7/dist-packages (from Flask>=0.8->flask-ngrok) (1.0.1)\n","Requirement already satisfied: Jinja2>=2.10.1 in /usr/local/lib/python3.7/dist-packages (from Flask>=0.8->flask-ngrok) (2.11.3)\n","Requirement already satisfied: click>=5.1 in /usr/local/lib/python3.7/dist-packages (from Flask>=0.8->flask-ngrok) (7.1.2)\n","Requirement already satisfied: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /usr/local/lib/python3.7/dist-packages (from requests->flask-ngrok) (1.24.3)\n","Requirement already satisfied: idna<3,>=2.5 in /usr/local/lib/python3.7/dist-packages (from requests->flask-ngrok) (2.10)\n","Requirement already satisfied: chardet<4,>=3.0.2 in /usr/local/lib/python3.7/dist-packages (from requests->flask-ngrok) (3.0.4)\n","Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.7/dist-packages (from requests->flask-ngrok) (2020.12.5)\n","Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.7/dist-packages (from Jinja2>=2.10.1->Flask>=0.8->flask-ngrok) (1.1.1)\n","Installing collected packages: flask-ngrok\n","Successfully installed flask-ngrok-0.0.25\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"iKtt8MMy9O0z","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1620213784888,"user_tz":-330,"elapsed":10048,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"c3a89848-60c2-4eee-faa1-0435453f078b"},"source":["!pip install dask[complete]"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Requirement already satisfied: dask[complete] in /usr/local/lib/python3.7/dist-packages (2.12.0)\n","Requirement already satisfied: toolz>=0.7.3; extra == \"complete\" in /usr/local/lib/python3.7/dist-packages (from dask[complete]) (0.11.1)\n","Requirement already satisfied: bokeh>=1.0.0; extra == \"complete\" in /usr/local/lib/python3.7/dist-packages (from dask[complete]) (2.3.1)\n","Requirement already satisfied: numpy>=1.13.0; extra == \"complete\" in /usr/local/lib/python3.7/dist-packages (from dask[complete]) (1.19.5)\n","Requirement already satisfied: PyYaml; extra == \"complete\" in /usr/local/lib/python3.7/dist-packages (from dask[complete]) (3.13)\n","Requirement already satisfied: pandas>=0.23.0; extra == \"complete\" in /usr/local/lib/python3.7/dist-packages (from dask[complete]) (1.1.5)\n","Requirement already satisfied: cloudpickle>=0.2.1; extra == \"complete\" in /usr/local/lib/python3.7/dist-packages (from dask[complete]) (1.3.0)\n","Collecting partd>=0.3.10; extra == \"complete\"\n","  Downloading https://files.pythonhosted.org/packages/41/94/360258a68b55f47859d72b2d0b2b3cfe0ca4fbbcb81b78812bd00ae86b7c/partd-1.2.0-py3-none-any.whl\n","Collecting fsspec>=0.6.0; extra == \"complete\"\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/e9/91/2ef649137816850fa4f4c97c6f2eabb1a79bf0aa2c8ed198e387e373455e/fsspec-2021.4.0-py3-none-any.whl (108kB)\n","\u001b[K     |████████████████████████████████| 112kB 4.9MB/s \n","\u001b[?25hCollecting distributed>=2.0; extra == \"complete\"\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/63/f8/ac2c18adde6477bca3881c4d3cfa74c7f4da7ee82f3c83c201aa3b9ca5ee/distributed-2021.4.1-py3-none-any.whl (696kB)\n","\u001b[K     |████████████████████████████████| 706kB 6.2MB/s \n","\u001b[?25hRequirement already satisfied: pillow>=7.1.0 in /usr/local/lib/python3.7/dist-packages (from bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (7.1.2)\n","Requirement already satisfied: Jinja2>=2.7 in /usr/local/lib/python3.7/dist-packages (from bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (2.11.3)\n","Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.7/dist-packages (from bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (2.8.1)\n","Requirement already satisfied: packaging>=16.8 in /usr/local/lib/python3.7/dist-packages (from bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (20.9)\n","Requirement already satisfied: tornado>=5.1 in /usr/local/lib/python3.7/dist-packages (from bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (5.1.1)\n","Requirement already satisfied: typing-extensions>=3.7.4 in /usr/local/lib/python3.7/dist-packages (from bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (3.7.4.3)\n","Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.7/dist-packages (from pandas>=0.23.0; extra == \"complete\"->dask[complete]) (2018.9)\n","Collecting locket\n","  Downloading https://files.pythonhosted.org/packages/50/b8/e789e45b9b9c2db75e9d9e6ceb022c8d1d7e49b2c085ce8c05600f90a96b/locket-0.2.1-py2.py3-none-any.whl\n","Requirement already satisfied: zict>=0.1.3 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (2.0.0)\n","Requirement already satisfied: setuptools in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (56.0.0)\n","Requirement already satisfied: sortedcontainers!=2.0.0,!=2.0.1 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (2.3.0)\n","Requirement already satisfied: tblib>=1.6.0 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (1.7.0)\n","Requirement already satisfied: psutil>=5.0 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (5.4.8)\n","Requirement already satisfied: msgpack>=0.6.0 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (1.0.2)\n","Requirement already satisfied: click>=6.6 in /usr/local/lib/python3.7/dist-packages (from distributed>=2.0; extra == \"complete\"->dask[complete]) (7.1.2)\n","Requirement already satisfied: MarkupSafe>=0.23 in /usr/local/lib/python3.7/dist-packages (from Jinja2>=2.7->bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (1.1.1)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.7/dist-packages (from python-dateutil>=2.1->bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (1.15.0)\n","Requirement already satisfied: pyparsing>=2.0.2 in /usr/local/lib/python3.7/dist-packages (from packaging>=16.8->bokeh>=1.0.0; extra == \"complete\"->dask[complete]) (2.4.7)\n","Requirement already satisfied: heapdict in /usr/local/lib/python3.7/dist-packages (from zict>=0.1.3->distributed>=2.0; extra == \"complete\"->dask[complete]) (1.0.1)\n","\u001b[31mERROR: distributed 2021.4.1 has requirement cloudpickle>=1.5.0, but you'll have cloudpickle 1.3.0 which is incompatible.\u001b[0m\n","\u001b[31mERROR: distributed 2021.4.1 has requirement dask>=2021.03.0, but you'll have dask 2.12.0 which is incompatible.\u001b[0m\n","Installing collected packages: locket, partd, fsspec, distributed\n","  Found existing installation: distributed 1.25.3\n","    Uninstalling distributed-1.25.3:\n","      Successfully uninstalled distributed-1.25.3\n","Successfully installed distributed-2021.4.1 fsspec-2021.4.0 locket-0.2.1 partd-1.2.0\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"13FWeU7Y9fls","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1620213788777,"user_tz":-330,"elapsed":13934,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"a3431545-3c90-4abb-89db-a14f83f55f8d"},"source":["pip install Flask-WTF"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Collecting Flask-WTF\n","  Downloading https://files.pythonhosted.org/packages/36/a9/8c01171066bd7a524ee005d81bb4a8aa446ab178043a1ad6cb5dc8f0bd83/Flask_WTF-0.14.3-py2.py3-none-any.whl\n","Requirement already satisfied: Flask in /usr/local/lib/python3.7/dist-packages (from Flask-WTF) (1.1.2)\n","Collecting WTForms\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/e0/31/614fc7dc7d76005b0acb8c0c8920d962b83d7422b4ba912886dfb63f86ff/WTForms-2.3.3-py2.py3-none-any.whl (169kB)\n","\u001b[K     |████████████████████████████████| 174kB 4.5MB/s \n","\u001b[?25hRequirement already satisfied: itsdangerous in /usr/local/lib/python3.7/dist-packages (from Flask-WTF) (1.1.0)\n","Requirement already satisfied: Werkzeug>=0.15 in /usr/local/lib/python3.7/dist-packages (from Flask->Flask-WTF) (1.0.1)\n","Requirement already satisfied: click>=5.1 in /usr/local/lib/python3.7/dist-packages (from Flask->Flask-WTF) (7.1.2)\n","Requirement already satisfied: Jinja2>=2.10.1 in /usr/local/lib/python3.7/dist-packages (from Flask->Flask-WTF) (2.11.3)\n","Requirement already satisfied: MarkupSafe in /usr/local/lib/python3.7/dist-packages (from WTForms->Flask-WTF) (1.1.1)\n","Installing collected packages: WTForms, Flask-WTF\n","Successfully installed Flask-WTF-0.14.3 WTForms-2.3.3\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"I5pJyB8U8uFH","executionInfo":{"status":"ok","timestamp":1620213825540,"user_tz":-330,"elapsed":50680,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"cb75bdfc-d755-4b06-82dc-4a0dbcc6ebcc"},"source":["from flask_ngrok import run_with_ngrok\n","from flask import Flask,render_template,session,url_for,redirect,request\n","import numpy as np \n","import os\n","import werkzeug\n","from nltk.stem import PorterStemmer\n","from joblib import load\n","import pickle\n","import os ,os.path\n","from pandas import DataFrame\n","import pandas as pd\n","import dask.dataframe as dd\n","import csv\n","from flask_wtf import FlaskForm\n","from wtforms import StringField,SubmitField,RadioField\n","from flask_wtf.file import FileField, FileRequired, FileAllowed\n","\n","# ================================================================================================\n","import spacy\n","\n","import string\n","from spacy.lang.en.stop_words import STOP_WORDS\n","from spacy.lang.en import English\n","from sklearn.feature_extraction.text import CountVectorizer,TfidfVectorizer\n","import en_core_web_sm\n","\n","# Create our list of punctuation marks\n","punctuations = string.punctuation\n","\n","# Create our list of stopwords\n","nlp = en_core_web_sm.load()\n","stop_words = spacy.lang.en.stop_words.STOP_WORDS\n","\n","# Load English tokenizer, tagger, parser, NER and word vectors\n","parser = English()\n","\n","from sklearn.base import TransformerMixin\n","\n","def spacy_tokenizer(sentence):\n","    # Creating our token object, which is used to create documents with linguistic annotations.\n","    mytokens = parser(sentence)\n","\n","    # Lemmatizing each token and converting each token into lowercase\n","    mytokens = [ word.lemma_.lower().strip() if word.lemma_ != \"-PRON-\" else word.lower_ for word in mytokens ]\n","\n","    # Removing stop words\n","    mytokens = [ word for word in mytokens if word not in stop_words and word not in punctuations ]\n","\n","    # return preprocessed list of tokens\n","    return mytokens\n","\n","class predictors(TransformerMixin):\n","    def transform(self, X, **transform_params):\n","        # Cleaning Text\n","        return [clean_text(text) for text in X]\n","\n","    def fit(self, X, y=None, **fit_params):\n","        return self\n","\n","    def get_params(self, deep=True):\n","        return {}\n","\n","# Basic function to clean the text\n","def clean_text(text):\n","    # Removing spaces and converting text into lowercase\n","    return text.strip().lower()\n","\n","tfidf_vector = TfidfVectorizer(tokenizer = spacy_tokenizer)\n","bow_vector = CountVectorizer(tokenizer = spacy_tokenizer, ngram_range=(1,1))\n","\n","# ====================================================================================\n","import tensorflow as tf\n","from tensorflow.keras.models import load_model\n","model = load_model(\"lstm_model.h5\")\n","\n","import tempfile\n","\n","model_dir = tempfile.gettempdir() # gets the path of the temporary folder\n","version = 1 # version of the specific model\n","export_path = os.path.join(model_dir, str(version))\n","if os.path.isdir(export_path):\n","  print(\"Found an existing version of the model. Deleting the previous version\\n\")\n","  !rm -r {export_path} # terminal command to remove a directory\n","\n","# Saving the model\n","tf.keras.models.save_model(\n","  model, # the trained model\n","  export_path, # export path for the model\n","  overwrite = True, # overwrite if existing version found\n","  include_optimizer = True, # include the model optimizer\n","  save_format = None,\n","  options = None\n",")\n","\n","# If tf.keras.models.save_model() doesn't work, use the model.save function\n","model.save(export_path, save_format=\"tf\")\n","             \n","print(\"Model Saved\\n\")\n","\n","os.environ['MODEL_DIR'] = model_dir\n","\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["WARNING:tensorflow:Layer lstm will not use cuDNN kernel since it doesn't meet the cuDNN kernel criteria. It will use generic GPU kernel as fallback when running on GPU\n","WARNING:tensorflow:Layer lstm will not use cuDNN kernel since it doesn't meet the cuDNN kernel criteria. It will use generic GPU kernel as fallback when running on GPU\n","WARNING:tensorflow:Layer lstm will not use cuDNN kernel since it doesn't meet the cuDNN kernel criteria. It will use generic GPU kernel as fallback when running on GPU\n","INFO:tensorflow:Assets written to: /tmp/1/assets\n","INFO:tensorflow:Assets written to: /tmp/1/assets\n","Model Saved\n","\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"6uKWMWOMI7-n"},"source":["import sys\n","# We need sudo prefix if not on a Google Colab.\n","if 'google.colab' not in sys.modules:\n","  SUDO_IF_NEEDED = 'sudo'\n","else:\n","  SUDO_IF_NEEDED = ''"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"wZAkgUQeI9zn","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1620213837111,"user_tz":-330,"elapsed":62247,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"3b040251-306f-41e7-e92a-36f92c3fda08"},"source":["!echo \"deb http://storage.googleapis.com/tensorflow-serving-apt stable tensorflow-model-server tensorflow-model-server-universal\" | {SUDO_IF_NEEDED} tee /etc/apt/sources.list.d/tensorflow-serving.list && \\\n","curl https://storage.googleapis.com/tensorflow-serving-apt/tensorflow-serving.release.pub.gpg | {SUDO_IF_NEEDED} apt-key add -\n","!{SUDO_IF_NEEDED} apt update"],"execution_count":null,"outputs":[{"output_type":"stream","text":["deb http://storage.googleapis.com/tensorflow-serving-apt stable tensorflow-model-server tensorflow-model-server-universal\n","  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n","                                 Dload  Upload   Total   Spent    Left  Speed\n","100  2943  100  2943    0     0   136k      0 --:--:-- --:--:-- --:--:--  136k\n","OK\n","Get:1 http://storage.googleapis.com/tensorflow-serving-apt stable InRelease [3,012 B]\n","Get:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease [3,626 B]\n","Get:3 http://storage.googleapis.com/tensorflow-serving-apt stable/tensorflow-model-server amd64 Packages [340 B]\n","Get:4 http://storage.googleapis.com/tensorflow-serving-apt stable/tensorflow-model-server-universal amd64 Packages [348 B]\n","Ign:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease\n","Get:6 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ Packages [53.9 kB]\n","Ign:7 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n","Get:8 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release [697 B]\n","Hit:9 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n","Get:10 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release.gpg [836 B]\n","Get:12 http://security.ubuntu.com/ubuntu bionic-security InRelease [88.7 kB]\n","Hit:13 http://archive.ubuntu.com/ubuntu bionic InRelease\n","Get:14 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease [15.9 kB]\n","Get:15 http://archive.ubuntu.com/ubuntu bionic-updates InRelease [88.7 kB]\n","Ign:16 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages\n","Get:16 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Packages [770 kB]\n","Hit:17 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease\n","Get:18 http://security.ubuntu.com/ubuntu bionic-security/multiverse amd64 Packages [24.7 kB]\n","Get:19 http://archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]\n","Get:20 http://security.ubuntu.com/ubuntu bionic-security/universe amd64 Packages [1,410 kB]\n","Get:21 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease [15.9 kB]\n","Get:22 http://archive.ubuntu.com/ubuntu bionic-updates/universe amd64 Packages [2,181 kB]\n","Get:23 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease [15.9 kB]\n","Get:24 http://security.ubuntu.com/ubuntu bionic-security/restricted amd64 Packages [399 kB]\n","Get:25 http://security.ubuntu.com/ubuntu bionic-security/main amd64 Packages [2,119 kB]\n","Get:26 http://archive.ubuntu.com/ubuntu bionic-updates/multiverse amd64 Packages [31.6 kB]\n","Get:27 http://archive.ubuntu.com/ubuntu bionic-updates/restricted amd64 Packages [429 kB]\n","Get:28 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main Sources [1,759 kB]\n","Get:29 http://archive.ubuntu.com/ubuntu bionic-updates/main amd64 Packages [2,550 kB]\n","Get:30 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic/main amd64 Packages [900 kB]\n","Get:31 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic/main amd64 Packages [40.8 kB]\n","Get:32 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic/main amd64 Packages [53.2 kB]\n","Fetched 13.0 MB in 9s (1,521 kB/s)\n","Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","85 packages can be upgraded. Run 'apt list --upgradable' to see them.\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"qH1PAdGkJIlk","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1620213849461,"user_tz":-330,"elapsed":74594,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"a3569e72-b797-4bc9-c7ac-88bc3634fcd7"},"source":["!{SUDO_IF_NEEDED} apt-get install tensorflow-model-server"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Reading package lists... Done\n","Building dependency tree       \n","Reading state information... Done\n","The following package was automatically installed and is no longer required:\n","  libnvidia-common-460\n","Use 'apt autoremove' to remove it.\n","The following NEW packages will be installed:\n","  tensorflow-model-server\n","0 upgraded, 1 newly installed, 0 to remove and 85 not upgraded.\n","Need to get 326 MB of archives.\n","After this operation, 0 B of additional disk space will be used.\n","Get:1 http://storage.googleapis.com/tensorflow-serving-apt stable/tensorflow-model-server amd64 tensorflow-model-server all 2.5.0 [326 MB]\n","Fetched 326 MB in 6s (54.7 MB/s)\n","Selecting previously unselected package tensorflow-model-server.\n","(Reading database ... 160690 files and directories currently installed.)\n","Preparing to unpack .../tensorflow-model-server_2.5.0_all.deb ...\n","Unpacking tensorflow-model-server (2.5.0) ...\n","Setting up tensorflow-model-server (2.5.0) ...\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"CziAHUYJGzAD","executionInfo":{"status":"ok","timestamp":1620213849462,"user_tz":-330,"elapsed":74582,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"d34599ea-7dd3-4bee-bac1-efdd005fa635"},"source":["%%bash --bg \n","nohup tensorflow_model_server \\\n","  --port=8500\\\n","  --rest_api_port=8501 \\\n","  --model_name=lstm_model \\\n","  --model_base_path=\"${MODEL_DIR}\" >server.log 2>&1\n"],"execution_count":null,"outputs":[{"output_type":"stream","text":["Starting job # 0 in a separate thread.\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"ncTjcG1qH_FE"},"source":["!tail server.log"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"b13v8c27H9wy","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1620213981051,"user_tz":-330,"elapsed":206156,"user":{"displayName":"Soumik Mukherjee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14GgBR34zaq7ReAnBqEKJjhUZbw4C1smq8hcBjN-wqQ=s64","userId":"15401072294633714494"}},"outputId":"1f666e90-5bba-4df5-d1b0-60c07f5f9d18"},"source":["from keras.preprocessing.text import Tokenizer\n","from keras.preprocessing.sequence import pad_sequences\n","import json\n","import requests\n","from tensorflow.keras.models import load_model\n","\n","tokenizer = Tokenizer()\n","\n","def decode_sentiment(score):\n","    return \"Business Exception\" if score>0.5 else \"System Exception\"\n","\n","def lstmPredict(exceptions):\n","  tokenizer.fit_on_texts(exceptions)\n","  processed = pad_sequences(tokenizer.texts_to_sequences(exceptions),maxlen =30)\n","  data = json.dumps({\"signature_name\": \"serving_default\", \"instances\": processed.tolist()})\n","  headers = {\"content-type\": \"application/json\"}\n","  json_response = requests.post('http://localhost:8051/v1/models/lstm_model:predict', data=data, headers=headers)\n","  print(json_response)\n","  preds = list(json.loads(json_response.text)['predictions'])\n","  processed = [decode_sentiment(score) for score in preds]\n","  lines = list()\n","  for i in range(len(processed)):\n","    line = list()\n","    line.append(processed[i])\n","    line.append(exceptions[i])\n","    s=\" : \"\n","    s= s.join(line)\n","    exceptions[i] = s\n","\n","\n","\t\n","\n","model_file_name = 'model.pkl' \n","def piplinePredict(model,exceptions):\n","\tops = model.predict(exceptions)\n","\tlines = list()\n","\tfor i in range(len(ops)):\n","\t\tline = list()\n","\t\tline.append(ops[i])\n","\t\tline.append(exceptions[i])\n","\t\ts=\" : \"\n","\t\ts= s.join(line)\n","\t\texceptions[i] = s\n","\n","\n","def stemming(wds):\n","\tporter = PorterStemmer()\n","\tfor i in range(len(wds)):\n","\t\twds[i] = porter.stem(wds[i])\n","\n","key_bus = ['contain','error','fields','mandatory','missing','found','template','file','field', 'input','processed','mail','invite','manually','id']\n","key_sys = ['internal','exception','memory','expression','code','stage','site','perform','evaluate','cleanup','time','hresult','failed','operation','unable','issue','data','fetch','unexpected','automatically','set','launch']\n","\n","porter = PorterStemmer()\n","def brute_force_audit(txts):\n","\tstemming(key_bus)\n","\tstemming(key_sys)\n","\tfor i in range(len(txts)):\n","\t\tsent = txts[i].lower()\n","\t\ttxts[i] = txts[i].split()\n","\t\twords = sent.split()\n","\t\tstemming(words)\n","\t\tBis_check =  any(item in key_bus for item in words)\n","\t\tSys_check =  any(item in key_sys for item in words)\n","\t\tif Bis_check == True:\n","\t\t\top = ['Business Exception : ']\n","\t\t\tfor j in txts[i]:\n","\t\t\t\top.append(j)\n","\t\t\ts=\" \"\n","\t\t\ts = s.join(op)\n","\t\t\ttxts[i] = s\n","\t\telif Sys_check == True:\n","\t\t\top = ['System Exception : ']\n","\t\t\tfor j in txts[i]:\n","\t\t\t\top.append(j)\n","\t\t\ts=\" \"\n","\t\t\ts = s.join(op)\n","\t\t\ttxts[i] = s\n","\n","\n","app = Flask(__name__)\n","app.config['SECRET_KEY'] = 'nlp_classifier'\n","run_with_ngrok(app) \n","\n","\n","@app.route(\"/\",methods = ['GET','POST'] )\n","def index():\n","\t\n","\tdat = list()\n","\top=list()\n","\tif request.method =='POST':\n","\t\tf = request.files['csvfile']\n","\t\tcf = isinstance(f,werkzeug.datastructures.FileStorage)\n","\t\tif f.filename!='':\n","\t\t\tf.save(os.path.join(os.getcwd(),f.filename))\n","\t\t\tfname = str(f.filename)\n","\t\t\tif fname !=\"\":\n","\t\t\t\tdf = dd.read_csv(fname)\n","\t\t\t\tdf = df.iloc[:,0].to_frame()\n","\t\t\t\tdf.to_csv('real_input.csv')\n","\n","\t\t\t\tDIR = str(os.getcwd())+\"/real_input.csv\"\n","\t\t\t\tnum =  (len([name for name in os.listdir(DIR) if os.path.isfile(os.path.join(DIR, name))]))\n","\t\t\t\tfor i in range(num):\n","  \t\t \t\t\tdf1 = pd.read_csv(DIR+'/'+str(i)+'.part')\n","  \t\t \t\t\tdat = dat + list(df1.iloc[:,1])\n","\n","\t\t\t\tana_type = request.form.getlist('Class')\n","\t\t\t\tif '3' in ana_type:\n","\t\t\t\t\tbrute_force_audit(dat)\n","\t\t\t\t\topdf = DataFrame(dat,columns=['Brute Force Results'])\n","\t\t\t\t\topdf.to_csv('brute_force_audit_results.csv')\n","\t\t\t\telif '2' in ana_type:\n","\t\t\t\t\tmodel = load(model_file_name)\n","\t\t\t\t\tpiplinePredict(model,dat)\n","\t\t\t\t\top_pip_df = DataFrame(dat,columns=['Spacy Classifier Results'])\n","\t\t\t\t\top_pip_df.to_csv('spacy_audit_results.csv')\n","\t\t\t\telif '1' in ana_type:\n","\t\t\t\t\tlstmPredict(dat)\n","\t\t\t\t\top_lstm_df = DataFrame(dat,columns=['LSTM Classifier Results'])\n","\t\t\t\t\top_lstm_df.to_csv('lstm_audit_results.csv')\n","\n","\t\t\t\t\t\n","\t\t\t\t\t\n","\t\t\telse:\n","\t\t\t\tdat.clear()\n","\t\telse:\n","\t\t\tdat.clear()\n","\t\t\n","\t\t# check for type of analysis\n","\t\t# pass the info in dat to that prediction function\n","\n","\n","\t\n","\treturn render_template('class_home.html',data = dat)\t\n","\n","\n","if __name__== '__main__':\n","\t\n","\tapp.run()"],"execution_count":null,"outputs":[{"output_type":"stream","text":[" * Serving Flask app \"__main__\" (lazy loading)\n"," * Environment: production\n","\u001b[31m   WARNING: This is a development server. Do not use it in a production deployment.\u001b[0m\n","\u001b[2m   Use a production WSGI server instead.\u001b[0m\n"," * Debug mode: off\n"],"name":"stdout"},{"output_type":"stream","text":[" * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)\n"],"name":"stderr"},{"output_type":"stream","text":[" * Running on http://a68ec9b0ebe8.ngrok.io\n"," * Traffic stats available on http://127.0.0.1:4040\n"],"name":"stdout"},{"output_type":"stream","text":["127.0.0.1 - - [05/May/2021 11:24:56] \"\u001b[37mGET / HTTP/1.1\u001b[0m\" 200 -\n","127.0.0.1 - - [05/May/2021 11:24:58] \"\u001b[33mGET /favicon.ico HTTP/1.1\u001b[0m\" 404 -\n","[2021-05-05 11:25:08,257] ERROR in app: Exception on / [POST]\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/connection.py\", line 159, in _new_conn\n","    (self._dns_host, self.port), self.timeout, **extra_kw)\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/util/connection.py\", line 80, in create_connection\n","    raise err\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/util/connection.py\", line 70, in create_connection\n","    sock.connect(sa)\n","ConnectionRefusedError: [Errno 111] Connection refused\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/connectionpool.py\", line 600, in urlopen\n","    chunked=chunked)\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/connectionpool.py\", line 354, in _make_request\n","    conn.request(method, url, **httplib_request_kw)\n","  File \"/usr/lib/python3.7/http/client.py\", line 1277, in request\n","    self._send_request(method, url, body, headers, encode_chunked)\n","  File \"/usr/lib/python3.7/http/client.py\", line 1323, in _send_request\n","    self.endheaders(body, encode_chunked=encode_chunked)\n","  File \"/usr/lib/python3.7/http/client.py\", line 1272, in endheaders\n","    self._send_output(message_body, encode_chunked=encode_chunked)\n","  File \"/usr/lib/python3.7/http/client.py\", line 1032, in _send_output\n","    self.send(msg)\n","  File \"/usr/lib/python3.7/http/client.py\", line 972, in send\n","    self.connect()\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/connection.py\", line 181, in connect\n","    conn = self._new_conn()\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/connection.py\", line 168, in _new_conn\n","    self, \"Failed to establish a new connection: %s\" % e)\n","urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f5901af10d0>: Failed to establish a new connection: [Errno 111] Connection refused\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.7/dist-packages/requests/adapters.py\", line 449, in send\n","    timeout=timeout\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/connectionpool.py\", line 638, in urlopen\n","    _stacktrace=sys.exc_info()[2])\n","  File \"/usr/local/lib/python3.7/dist-packages/urllib3/util/retry.py\", line 399, in increment\n","    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n","urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8051): Max retries exceeded with url: /v1/models/lstm_model:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f5901af10d0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.7/dist-packages/flask/app.py\", line 2447, in wsgi_app\n","    response = self.full_dispatch_request()\n","  File \"/usr/local/lib/python3.7/dist-packages/flask/app.py\", line 1952, in full_dispatch_request\n","    rv = self.handle_user_exception(e)\n","  File \"/usr/local/lib/python3.7/dist-packages/flask/app.py\", line 1821, in handle_user_exception\n","    reraise(exc_type, exc_value, tb)\n","  File \"/usr/local/lib/python3.7/dist-packages/flask/_compat.py\", line 39, in reraise\n","    raise value\n","  File \"/usr/local/lib/python3.7/dist-packages/flask/app.py\", line 1950, in full_dispatch_request\n","    rv = self.dispatch_request()\n","  File \"/usr/local/lib/python3.7/dist-packages/flask/app.py\", line 1936, in dispatch_request\n","    return self.view_functions[rule.endpoint](**req.view_args)\n","  File \"<ipython-input-10-fab521351091>\", line 121, in index\n","    lstmPredict(dat)\n","  File \"<ipython-input-10-fab521351091>\", line 17, in lstmPredict\n","    json_response = requests.post('http://localhost:8051/v1/models/lstm_model:predict', data=data, headers=headers)\n","  File \"/usr/local/lib/python3.7/dist-packages/requests/api.py\", line 119, in post\n","    return request('post', url, data=data, json=json, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/requests/api.py\", line 61, in request\n","    return session.request(method=method, url=url, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/requests/sessions.py\", line 530, in request\n","    resp = self.send(prep, **send_kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/requests/sessions.py\", line 643, in send\n","    r = adapter.send(request, **kwargs)\n","  File \"/usr/local/lib/python3.7/dist-packages/requests/adapters.py\", line 516, in send\n","    raise ConnectionError(e, request=request)\n","requests.exceptions.ConnectionError: HTTPConnectionPool(host='localhost', port=8051): Max retries exceeded with url: /v1/models/lstm_model:predict (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f5901af10d0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n","127.0.0.1 - - [05/May/2021 11:25:08] \"\u001b[35m\u001b[1mPOST / HTTP/1.1\u001b[0m\" 500 -\n"],"name":"stderr"}]},{"cell_type":"code","metadata":{"id":"P2GGLY9f5vEg"},"source":[""],"execution_count":null,"outputs":[]}]}